src/pages/Graph/SummaryPanelNode.tsx

```diff
@@ -117,19 +132,53 @@ export class SummaryPanelNode extends React.Component<SummaryPanelNodeProps, Sum
             {shouldRenderSvcList && <div>{servicesList}</div>}
             {shouldRenderService && <div>{renderBadgedLink(nodeData, NodeType.SERVICE)}</div>}
             {shouldRenderApp && <div>{renderBadgedLink(nodeData, NodeType.APP)}</div>}
-            {shouldRenderWorkload && <div>{renderBadgedLink(nodeData, NodeType.WORKLOAD)}</div>}
+            {shouldRenderWorkload && this.renderWorkloadSection(nodeData)}
           </div>
         </div>
         {shouldRenderTraces ? this.renderWithTabs(nodeData) : this.renderTrafficOnly()}
       </div>
     );
   }
 
-  private renderGatewayHostnames = (nodeData:  DecoratedGraphNodeData) => {
+  private renderWorkloadSection = (nodeData: DecoratedGraphNodeData) => {
+    if (!nodeData.hasWorkloadEntry) {
+      return <div>{renderBadgedLink(nodeData, NodeType.WORKLOAD)}</div>;
+    }
+
+    const workloadEntryLinks = nodeData.hasWorkloadEntry.map(we => (
+      <div>
+        {getLink(nodeData, NodeType.WORKLOAD, () => ({
+          link: `/namespaces/${encodeURIComponent(nodeData.namespace)}/istio/workloadentries/${encodeURIComponent(
+            we.name
+          )}`,
+          displayName: we.name,
+          key: `${nodeData.namespace}.wle.${we.name}`
+        }))}
+      </div>
+    ));
+
+    return (
+      <>
+        {getBadge(nodeData, NodeType.WORKLOAD)}
+        <Expandable
+          toggleText={
+            nodeData.hasWorkloadEntry.length === 1
+              ? '1 workload entry'
+              : `${nodeData.hasWorkloadEntry.length} workload entries`
+          }
+          className={classes(expandableSectionStyle, style({ display: 'inline' }))}
```

Comment:
(Nitpick: performance?)

I wonder if it is better to declare that `style({ display: 'inline' })` in a `const`?

I'm not sure if `typestyle` is smart to reuse styles. If not, repeated invocations to `style` may be injecting several classNames to the document with the same css definition. 

I didn't check, so I cannot tell.

Reply:
> I wonder if it is better to declare that style({ display: 'inline' }) in a const?

:+1: 

> I'm not sure if typestyle is smart to reuse styles. If not, repeated invocations to style may be injecting several classNames to the document with the same css definition.

It looks like the [classes method](https://typestyle.github.io/#/core/classes) just adds the two styles as separate classes relying on css composition: e.g. `<p class="class1 class2">`. In this case the only thing defined in `class2` would be `{display: 'inline'}`. I'm happy to update if that's not desirable.

Reply:
> In this case the only thing defined in `class2` would be `{display: 'inline'}`. I'm happy to update if that's not desirable.

That's perfectly fine.

I understand you would see `<p class="class1 class2">` in the first call. However, since you are calling repeatedly the `style` function, I was wondering if you would see `<p class="class1 class3">`, then `<p class="class1 class4">`, ..., `<p class="class1 classN">` in subsequent calls/renders. This is what I think wouldn't be desirable. But if you always see `<p class="class1 class2">`, that's fine.

Reply:
Ah I gotcha, I'll pull that out so it's not inlined.